### ISSUE TITLE
[BUG] script stops after an hour or so

### ISSUE THREAD

**Issue by bananachicken** [rdeepak2002/reddit-place-script-2022#372]
it has never completed the image requested, but stops at a random point after a certain amount of time. seems to be crashing.

i'm using the latest version and they are placing the pixels correctly, but only for a certain period of time

**Comment by bjb28** [comment]
Can you provide any errors or log output?

**Comment by bananachicken** [comment]
aha! i just managed to grab one before it crashed.

`Exception in thread Thread-7:
Traceback (most recent call last):
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.8_3.8.2800.0_x64__qbz5n2kfra8p0\lib\threading.py", line 932, in _bootstrap_inner
    self.run()
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.8_3.8.2800.0_x64__qbz5n2kfra8p0\lib\threading.py", line 870, in run
    self._target(*self._args, **self._kwargs)
  File "main.py", line 702, in task
    current_r, current_c, new_rgb = self.get_unset_pixel(
  File "main.py", line 490, in get_unset_pixel
    boardimg = self.get_board(self.access_tokens[index])
  File "main.py", line 319, in get_board
    msg = ws.recv()
  File "C:\Users\mabel\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.8_qbz5n2kfra8p0\LocalCache\local-packages\Python38\site-packages\websocket\_core.py", line 357, in recv
    opcode, data = self.recv_data()
  File "C:\Users\mabel\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.8_qbz5n2kfra8p0\LocalCache\local-packages\Python38\site-packages\websocket\_core.py", line 380, in recv_data
    opcode, frame = self.recv_data_frame(control_frame)
  File "C:\Users\mabel\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.8_qbz5n2kfra8p0\LocalCache\local-packages\Python38\site-packages\websocket\_core.py", line 401, in recv_data_frame
    frame = self.recv_frame()
  File "C:\Users\mabel\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.8_qbz5n2kfra8p0\LocalCache\local-packages\Python38\site-packages\websocket\_core.py", line 440, in recv_frame
    return self.frame_buffer.recv_frame()
  File "C:\Users\mabel\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.8_qbz5n2kfra8p0\LocalCache\local-packages\Python38\site-packages\websocket\_abnf.py", line 338, in recv_frame
    self.recv_header()
  File "C:\Users\mabel\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.8_qbz5n2kfra8p0\LocalCache\local-packages\Python38\site-packages\websocket\_abnf.py", line 294, in recv_header
    header = self.recv_strict(2)
  File "C:\Users\mabel\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.8_qbz5n2kfra8p0\LocalCache\local-packages\Python38\site-packages\websocket\_abnf.py", line 373, in recv_strict
    bytes_ = self.recv(min(16384, shortage))
  File "C:\Users\mabel\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.8_qbz5n2kfra8p0\LocalCache\local-packages\Python38\site-packages\websocket\_core.py", line 524, in _recv
    return recv(self.sock, bufsize)
  File "C:\Users\mabel\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.8_qbz5n2kfra8p0\LocalCache\local-packages\Python38\site-packages\websocket\_socket.py", line 122, in recv
    raise WebSocketConnectionClosedException(
websocket._exceptions.WebSocketConnectionClosedException: Connection to remote host was lost.
`

sorry for the jumble of text, lol

**Comment by omar2205** [comment]
The WebSocket drops the connection after a while, we should handle `WebSocketConnectionClosedException` and retry

**Comment by bananachicken** [comment]
update: it seems to crash when all of the accounts are rate-limited instead of retrying. left it on overnight and it barely got through an hour

**Comment by toastythetoaster** [comment]
I am also getting the same thing while all of the workers are rate-limited

**Comment by InertiaImpact** [comment]
I got the same issue (forgot to grab error message) but none of my workers were rate limited. All were verified.

**Comment by InertiaImpact** [comment]
> I got the same issue (forgot to grab error message) but none of my workers were rate limited. All were verified.

`Exception in thread Thread-2:
Traceback (most recent call last):
  File "C:\Python39\lib\threading.py", line 973, in _bootstrap_inner
    self.run()
  File "C:\Python39\lib\threading.py", line 910, in run
    self._target(*self._args, **self._kwargs)
  File "C:\Users\Plex\Downloads\reddit-place-script-2022-main\reddit-place-script-2022-main\main.py", line 606, in task
    next_pixel_placement_time = self.set_pixel_and_check_ratelimit(
  File "C:\Users\Plex\Downloads\reddit-place-script-2022-main\reddit-place-script-2022-main\main.py", line 135, in set_pixel_and_check_ratelimit
    if response.json()["data"] is None:
KeyError: 'data'
Exception in thread Thread-1:
Traceback (most recent call last):
  File "C:\Python39\lib\threading.py", line 973, in _bootstrap_inner
    self.run()
  File "C:\Python39\lib\threading.py", line 910, in run
    self._target(*self._args, **self._kwargs)
  File "C:\Users\Plex\Downloads\reddit-place-script-2022-main\reddit-place-script-2022-main\main.py", line 606, in task
    next_pixel_placement_time = self.set_pixel_and_check_ratelimit(
  File "C:\Users\Plex\Downloads\reddit-place-script-2022-main\reddit-place-script-2022-main\main.py", line 135, in set_pixel_and_check_ratelimit
    if response.json()["data"] is None:
KeyError: 'data'
Exception in thread Thread-7:
Traceback (most recent call last):
  File "C:\Python39\lib\threading.py", line 973, in _bootstrap_inner
    self.run()
  File "C:\Python39\lib\threading.py", line 910, in run
    self._target(*self._args, **self._kwargs)
  File "C:\Users\Plex\Downloads\reddit-place-script-2022-main\reddit-place-script-2022-main\main.py", line 581, in task
    current_r, current_c, new_rgb = self.get_unset_pixel(
  File "C:\Users\Plex\Downloads\reddit-place-script-2022-main\reddit-place-script-2022-main\main.py", line 358, in get_unset_pixel
    boardimg = self.get_board(self.access_tokens[index])
  File "C:\Users\Plex\Downloads\reddit-place-script-2022-main\reddit-place-script-2022-main\main.py", line 187, in get_board
    msg = ws.recv()
  File "C:\Python39\lib\site-packages\websocket\_core.py", line 357, in recv
    opcode, data = self.recv_data()
  File "C:\Python39\lib\site-packages\websocket\_core.py", line 380, in recv_data
    opcode, frame = self.recv_data_frame(control_frame)
  File "C:\Python39\lib\site-packages\websocket\_core.py", line 401, in recv_data_frame
    frame = self.recv_frame()
  File "C:\Python39\lib\site-packages\websocket\_core.py", line 440, in recv_frame
    return self.frame_buffer.recv_frame()
  File "C:\Python39\lib\site-packages\websocket\_abnf.py", line 338, in recv_frame
    self.recv_header()
  File "C:\Python39\lib\site-packages\websocket\_abnf.py", line 294, in recv_header
    header = self.recv_strict(2)
  File "C:\Python39\lib\site-packages\websocket\_abnf.py", line 373, in recv_strict
    bytes_ = self.recv(min(16384, shortage))
  File "C:\Python39\lib\site-packages\websocket\_core.py", line 524, in _recv
    return recv(self.sock, bufsize)
  File "C:\Python39\lib\site-packages\websocket\_socket.py", line 122, in recv
    raise WebSocketConnectionClosedException(
websocket._exceptions.WebSocketConnectionClosedException: Connection to remote host was lost.
Exception in thread Thread-6:
Traceback (most recent call last):
  File "C:\Python39\lib\threading.py", line 973, in _bootstrap_inner
    self.run()
  File "C:\Python39\lib\threading.py", line 910, in run
    self._target(*self._args, **self._kwargs)
  File "C:\Users\Plex\Downloads\reddit-place-script-2022-main\reddit-place-script-2022-main\main.py", line 581, in task
    current_r, current_c, new_rgb = self.get_unset_pixel(
  File "C:\Users\Plex\Downloads\reddit-place-script-2022-main\reddit-place-script-2022-main\main.py", line 358, in get_unset_pixel
    boardimg = self.get_board(self.access_tokens[index])
  File "C:\Users\Plex\Downloads\reddit-place-script-2022-main\reddit-place-script-2022-main\main.py", line 187, in get_board
    msg = ws.recv()
  File "C:\Python39\lib\site-packages\websocket\_core.py", line 357, in recv
    opcode, data = self.recv_data()
  File "C:\Python39\lib\site-packages\websocket\_core.py", line 380, in recv_data
    opcode, frame = self.recv_data_frame(control_frame)
  File "C:\Python39\lib\site-packages\websocket\_core.py", line 401, in recv_data_frame
    frame = self.recv_frame()
  File "C:\Python39\lib\site-packages\websocket\_core.py", line 440, in recv_frame
    return self.frame_buffer.recv_frame()
  File "C:\Python39\lib\site-packages\websocket\_abnf.py", line 338, in recv_frame
    self.recv_header()
  File "C:\Python39\lib\site-packages\websocket\_abnf.py", line 294, in recv_header
    header = self.recv_strict(2)
  File "C:\Python39\lib\site-packages\websocket\_abnf.py", line 373, in recv_strict
    bytes_ = self.recv(min(16384, shortage))
  File "C:\Python39\lib\site-packages\websocket\_core.py", line 524, in _recv
    return recv(self.sock, bufsize)
  File "C:\Python39\lib\site-packages\websocket\_socket.py", line 122, in recv
    raise WebSocketConnectionClosedException(
websocket._exceptions.WebSocketConnectionClosedException: Connection to remote host was lost.
Exception in thread Thread-3:
Traceback (most recent call last):
  File "C:\Python39\lib\threading.py", line 973, in _bootstrap_inner
    self.run()
  File "C:\Python39\lib\threading.py", line 910, in run
    self._target(*self._args, **self._kwargs)
  File "C:\Users\Plex\Downloads\reddit-place-script-2022-main\reddit-place-script-2022-main\main.py", line 581, in task
    current_r, current_c, new_rgb = self.get_unset_pixel(
  File "C:\Users\Plex\Downloads\reddit-place-script-2022-main\reddit-place-script-2022-main\main.py", line 358, in get_unset_pixel
    boardimg = self.get_board(self.access_tokens[index])
  File "C:\Users\Plex\Downloads\reddit-place-script-2022-main\reddit-place-script-2022-main\main.py", line 187, in get_board
    msg = ws.recv()
  File "C:\Python39\lib\site-packages\websocket\_core.py", line 357, in recv
    opcode, data = self.recv_data()
  File "C:\Python39\lib\site-packages\websocket\_core.py", line 380, in recv_data
    opcode, frame = self.recv_data_frame(control_frame)
  File "C:\Python39\lib\site-packages\websocket\_core.py", line 401, in recv_data_frame
    frame = self.recv_frame()
  File "C:\Python39\lib\site-packages\websocket\_core.py", line 440, in recv_frame
    return self.frame_buffer.recv_frame()
  File "C:\Python39\lib\site-packages\websocket\_abnf.py", line 338, in recv_frame
    self.recv_header()
  File "C:\Python39\lib\site-packages\websocket\_abnf.py", line 294, in recv_header
    header = self.recv_strict(2)
  File "C:\Python39\lib\site-packages\websocket\_abnf.py", line 373, in recv_strict
    bytes_ = self.recv(min(16384, shortage))
  File "C:\Python39\lib\site-packages\websocket\_core.py", line 524, in _recv
    return recv(self.sock, bufsize)
  File "C:\Python39\lib\site-packages\websocket\_socket.py", line 122, in recv
    raise WebSocketConnectionClosedException(
websocket._exceptions.WebSocketConnectionClosedException: Connection to remote host was lost.
Exception in thread Thread-11:
Traceback (most recent call last):
  File "C:\Python39\lib\threading.py", line 973, in _bootstrap_inner
    self.run()
  File "C:\Python39\lib\threading.py", line 910, in run
    self._target(*self._args, **self._kwargs)
  File "C:\Users\Plex\Downloads\reddit-place-script-2022-main\reddit-place-script-2022-main\main.py", line 581, in task
    current_r, current_c, new_rgb = self.get_unset_pixel(
  File "C:\Users\Plex\Downloads\reddit-place-script-2022-main\reddit-place-script-2022-main\main.py", line 358, in get_unset_pixel
    boardimg = self.get_board(self.access_tokens[index])
  File "C:\Users\Plex\Downloads\reddit-place-script-2022-main\reddit-place-script-2022-main\main.py", line 187, in get_board
    msg = ws.recv()
  File "C:\Python39\lib\site-packages\websocket\_core.py", line 357, in recv
    opcode, data = self.recv_data()
  File "C:\Python39\lib\site-packages\websocket\_core.py", line 380, in recv_data
    opcode, frame = self.recv_data_frame(control_frame)
  File "C:\Python39\lib\site-packages\websocket\_core.py", line 401, in recv_data_frame
    frame = self.recv_frame()
  File "C:\Python39\lib\site-packages\websocket\_core.py", line 440, in recv_frame
    return self.frame_buffer.recv_frame()
  File "C:\Python39\lib\site-packages\websocket\_abnf.py", line 338, in recv_frame
    self.recv_header()
  File "C:\Python39\lib\site-packages\websocket\_abnf.py", line 294, in recv_header
    header = self.recv_strict(2)
  File "C:\Python39\lib\site-packages\websocket\_abnf.py", line 373, in recv_strict
    bytes_ = self.recv(min(16384, shortage))
  File "C:\Python39\lib\site-packages\websocket\_core.py", line 524, in _recv
    return recv(self.sock, bufsize)
  File "C:\Python39\lib\site-packages\websocket\_socket.py", line 122, in recv
    raise WebSocketConnectionClosedException(
websocket._exceptions.WebSocketConnectionClosedException: Connection to remote host was lost.
Exception in thread Thread-4:
Traceback (most recent call last):
  File "C:\Python39\lib\threading.py", line 973, in _bootstrap_inner
    self.run()
  File "C:\Python39\lib\threading.py", line 910, in run
    self._target(*self._args, **self._kwargs)
  File "C:\Users\Plex\Downloads\reddit-place-script-2022-main\reddit-place-script-2022-main\main.py", line 581, in task
    current_r, current_c, new_rgb = self.get_unset_pixel(
  File "C:\Users\Plex\Downloads\reddit-place-script-2022-main\reddit-place-script-2022-main\main.py", line 358, in get_unset_pixel
    boardimg = self.get_board(self.access_tokens[index])
  File "C:\Users\Plex\Downloads\reddit-place-script-2022-main\reddit-place-script-2022-main\main.py", line 187, in get_board
    msg = ws.recv()
  File "C:\Python39\lib\site-packages\websocket\_core.py", line 357, in recv
    opcode, data = self.recv_data()
  File "C:\Python39\lib\site-packages\websocket\_core.py", line 380, in recv_data
    opcode, frame = self.recv_data_frame(control_frame)
  File "C:\Python39\lib\site-packages\websocket\_core.py", line 401, in recv_data_frame
    frame = self.recv_frame()
  File "C:\Python39\lib\site-packages\websocket\_core.py", line 440, in recv_frame
    return self.frame_buffer.recv_frame()
  File "C:\Python39\lib\site-packages\websocket\_abnf.py", line 338, in recv_frame
    self.recv_header()
  File "C:\Python39\lib\site-packages\websocket\_abnf.py", line 294, in recv_header
    header = self.recv_strict(2)
  File "C:\Python39\lib\site-packages\websocket\_abnf.py", line 373, in recv_strict
    bytes_ = self.recv(min(16384, shortage))
  File "C:\Python39\lib\site-packages\websocket\_core.py", line 524, in _recv
    return recv(self.sock, bufsize)
  File "C:\Python39\lib\site-packages\websocket\_socket.py", line 122, in recv
    raise WebSocketConnectionClosedException(
websocket._exceptions.WebSocketConnectionClosedException: Connection to remote host was lost.
Exception in thread Thread-12:
Traceback (most recent call last):
  File "C:\Python39\lib\threading.py", line 973, in _bootstrap_inner
    self.run()
  File "C:\Python39\lib\threading.py", line 910, in run
    self._target(*self._args, **self._kwargs)
  File "C:\Users\Plex\Downloads\reddit-place-script-2022-main\reddit-place-script-2022-main\main.py", line 581, in task
    current_r, current_c, new_rgb = self.get_unset_pixel(
  File "C:\Users\Plex\Downloads\reddit-place-script-2022-main\reddit-place-script-2022-main\main.py", line 358, in get_unset_pixel
    boardimg = self.get_board(self.access_tokens[index])
  File "C:\Users\Plex\Downloads\reddit-place-script-2022-main\reddit-place-script-2022-main\main.py", line 187, in get_board
    msg = ws.recv()
  File "C:\Python39\lib\site-packages\websocket\_core.py", line 357, in recv
    opcode, data = self.recv_data()
  File "C:\Python39\lib\site-packages\websocket\_core.py", line 380, in recv_data
    opcode, frame = self.recv_data_frame(control_frame)
  File "C:\Python39\lib\site-packages\websocket\_core.py", line 401, in recv_data_frame
    frame = self.recv_frame()
  File "C:\Python39\lib\site-packages\websocket\_core.py", line 440, in recv_frame
    return self.frame_buffer.recv_frame()
  File "C:\Python39\lib\site-packages\websocket\_abnf.py", line 338, in recv_frame
    self.recv_header()
  File "C:\Python39\lib\site-packages\websocket\_abnf.py", line 294, in recv_header
    header = self.recv_strict(2)
  File "C:\Python39\lib\site-packages\websocket\_abnf.py", line 373, in recv_strict
    bytes_ = self.recv(min(16384, shortage))
  File "C:\Python39\lib\site-packages\websocket\_core.py", line 524, in _recv
    return recv(self.sock, bufsize)
  File "C:\Python39\lib\site-packages\websocket\_socket.py", line 122, in recv
    raise WebSocketConnectionClosedException(
websocket._exceptions.WebSocketConnectionClosedException: Connection to remote host was lost.
Exception in thread Thread-5:
Traceback (most recent call last):
  File "C:\Python39\lib\threading.py", line 973, in _bootstrap_inner
    self.run()
  File "C:\Python39\lib\threading.py", line 910, in run
    self._target(*self._args, **self._kwargs)
  File "C:\Users\Plex\Downloads\reddit-place-script-2022-main\reddit-place-script-2022-main\main.py", line 581, in task
    current_r, current_c, new_rgb = self.get_unset_pixel(
  File "C:\Users\Plex\Downloads\reddit-place-script-2022-main\reddit-place-script-2022-main\main.py", line 358, in get_unset_pixel
    boardimg = self.get_board(self.access_tokens[index])
  File "C:\Users\Plex\Downloads\reddit-place-script-2022-main\reddit-place-script-2022-main\main.py", line 187, in get_board
    msg = ws.recv()
  File "C:\Python39\lib\site-packages\websocket\_core.py", line 357, in recv
    opcode, data = self.recv_data()
  File "C:\Python39\lib\site-packages\websocket\_core.py", line 380, in recv_data
    opcode, frame = self.recv_data_frame(control_frame)
  File "C:\Python39\lib\site-packages\websocket\_core.py", line 401, in recv_data_frame
    frame = self.recv_frame()
  File "C:\Python39\lib\site-packages\websocket\_core.py", line 440, in recv_frame
    return self.frame_buffer.recv_frame()
  File "C:\Python39\lib\site-packages\websocket\_abnf.py", line 338, in recv_frame
    self.recv_header()
  File "C:\Python39\lib\site-packages\websocket\_abnf.py", line 294, in recv_header
    header = self.recv_strict(2)
  File "C:\Python39\lib\site-packages\websocket\_abnf.py", line 373, in recv_strict
    bytes_ = self.recv(min(16384, shortage))
  File "C:\Python39\lib\site-packages\websocket\_core.py", line 524, in _recv
    return recv(self.sock, bufsize)
  File "C:\Python39\lib\site-packages\websocket\_socket.py", line 122, in recv
    raise WebSocketConnectionClosedException(
websocket._exceptions.WebSocketConnectionClosedException: Connection to remote host was lost.
Exception in thread Thread-8:
Traceback (most recent call last):
  File "C:\Python39\lib\threading.py", line 973, in _bootstrap_inner
    self.run()
  File "C:\Python39\lib\threading.py", line 910, in run
    self._target(*self._args, **self._kwargs)
  File "C:\Users\Plex\Downloads\reddit-place-script-2022-main\reddit-place-script-2022-main\main.py", line 581, in task
    current_r, current_c, new_rgb = self.get_unset_pixel(
  File "C:\Users\Plex\Downloads\reddit-place-script-2022-main\reddit-place-script-2022-main\main.py", line 358, in get_unset_pixel
    boardimg = self.get_board(self.access_tokens[index])
  File "C:\Users\Plex\Downloads\reddit-place-script-2022-main\reddit-place-script-2022-main\main.py", line 187, in get_board
    msg = ws.recv()
  File "C:\Python39\lib\site-packages\websocket\_core.py", line 357, in recv
    opcode, data = self.recv_data()
  File "C:\Python39\lib\site-packages\websocket\_core.py", line 380, in recv_data
    opcode, frame = self.recv_data_frame(control_frame)
  File "C:\Python39\lib\site-packages\websocket\_core.py", line 401, in recv_data_frame
    frame = self.recv_frame()
  File "C:\Python39\lib\site-packages\websocket\_core.py", line 440, in recv_frame
    return self.frame_buffer.recv_frame()
  File "C:\Python39\lib\site-packages\websocket\_abnf.py", line 338, in recv_frame
    self.recv_header()
  File "C:\Python39\lib\site-packages\websocket\_abnf.py", line 294, in recv_header
    header = self.recv_strict(2)
  File "C:\Python39\lib\site-packages\websocket\_abnf.py", line 373, in recv_strict
    bytes_ = self.recv(min(16384, shortage))
  File "C:\Python39\lib\site-packages\websocket\_core.py", line 524, in _recv
    return recv(self.sock, bufsize)
  File "C:\Python39\lib\site-packages\websocket\_socket.py", line 122, in recv
    raise WebSocketConnectionClosedException(
websocket._exceptions.WebSocketConnectionClosedException: Connection to remote host was lost.
`

**Comment by Adenser** [comment]
Seems like it's duplicate of #348

**Comment by Skyline-9** [comment]
Is the issue fixed after #428?
